### PR TITLE
Add auto-update using GitHub Actions

### PR DESCRIPTION
**Pull request by johnhaddon**
This adds a GitHub workflow that periodically checks for new releases on GafferHQ/gaffer, automatically opening a pull request to update the documentation when necessary. An example is https://github.com/GafferHQ/documentation/pull/14, which was created in response to me pushing the branch for this PR.

### CODE REVIEW

File: update.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+shutil.copytree(
+	"/tmp/{}/doc/gaffer/html".format( packageName ),
+	"./{}".format( __versionToString( latestReleaseVersion ) )
+)
+subprocess.call( [ "git", "add", "./{}".format( __versionToString( latestReleaseVersion ) ) ] )
+
+# Update index.html to redirect to latest version
+
+with open( "index.html.in" ) as indexIn :
```
Comment (themissingcow) [inline update.py]: fwiw, I think Python allows multiple context managers per `with` clause.

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      run: |
+        python --version
+        pip install PyGitHub==1.45
+
+    - name: Update
+      id: update
+      run: |
+        ./update.py --set-outputs 1
+
+    - name: Create Pull Request
```
Comment (themissingcow) [inline .github/workflows/update.yml:33]: Is this relying on `create-pull-request` to be a no-op if there are no changes?

File: index.html [hunk 3]
```diff
@@ -2,10 +2,10 @@
 <html lang="en-US">
     <meta charset="utf-8">
     <title>Redirecting&hellip;</title>
-    <link rel="canonical" href="0.57.5.0/index.html">
-    <script>location="0.57.5.0/index.html"</script>
-    <meta http-equiv="refresh" content="0; url=0.57.5.0/index.html">
+    <link rel="canonical" href="0.59.0.0/index.html">
```
Comment (themissingcow) [inline index.html:5]: Should this change be part of this PR?

File: update.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		continue
+
+	releaseVersion = __versionFromString( release.tag_name )
+	if releaseVersion > latestReleaseVersion :
+		latestRelease = release
+		latestReleaseVersion = releaseVersion
+
+# # If its up to date with the latest documented version, then we're done
+
+latestDocVersion = max( [ __versionFromString( v ) for v in glob.glob( "*.*.*.*" ) ] )
```
Comment (themissingcow) [inline update.py:66]: Given our past experience with failed deploys, Is it worth us checking how long `glob.glob(...)` is, and warning/erroring if there are more than _n_ versions already present.

**Review by themissingcow — COMMENTED** [review]
This is going to really help, nice one.

File: .github/workflows/update.yml (hunk 2 above)
Comment (johnhaddon) [reply .github/workflows/update.yml:33]: Correct, and that is the documented behaviour : 

> If there are no changes (i.e. no diff exists with the checked-out base branch), no pull request will be created and the action exits silently.

File: update.py (hunk 1 above)
Comment (johnhaddon) [reply update.py]: Used in 97a9c84.

File: index.html (hunk 3 above)
Comment (johnhaddon) [reply index.html:5]: It should not - thanks for spotting - removed in 97a9c84.

File: update.py (hunk 4 above)
Comment (johnhaddon) [reply update.py:66]: Definitely, but I deliberately didn't tackle that this time. I will automate next time we hit the problem, when we can figure out what `n` should be (I presume it's not versions, but overall number of files or file size).

**Review by themissingcow — APPROVED** [review]

**Pull request merged**